pkg/ingester/ingester_v2.go

```diff
@@ -847,6 +851,37 @@ func (i *Ingester) v2Push(ctx context.Context, req *cortexpb.WriteRequest) (*cor
 				return copiedLabels
 			})
 		}
+
+		// app.AppendExemplar currently doesn't create the series, it must
+		// already exist.  If it does not then drop.
+		if ref == 0 && len(ts.Exemplars) > 0 {
+			updateFirstPartial(func() error {
+				return wrappedTSDBIngestExemplarErr(errors.New("exemplars not ingested because series not already present"),
+					model.Time(ts.Exemplars[0].TimestampMs), ts.Labels, ts.Exemplars[0].Labels)
+			})
+			failedExemplarsCount += len(ts.Exemplars)
+			continue
```

Comment:
This `continue` could be error prone in the future, if other logic is added below. I think it's better to get rid of it and then the following logic `for _, ex := range ts.Exemplars {` in a `else` branch.